hummingbot/connector/exchange/wazirx/wazirx_order_book_message.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+from typing import (
+    Dict,
+    List,
+    Optional,
+)
+
+from hummingbot.core.data_type.order_book_row import OrderBookRow
+from hummingbot.core.data_type.order_book_message import (
+    OrderBookMessage,
+    OrderBookMessageType,
+)
+
+
+class WazirxOrderBookMessage(OrderBookMessage):
+    def __new__(
+        cls,
+        message_type: OrderBookMessageType,
+        content: Dict[str, any],
+        timestamp: Optional[float] = None,
+        *args,
+        **kwargs,
+    ):
+        if timestamp is None:
+            if message_type is OrderBookMessageType.SNAPSHOT:
+                raise ValueError("timestamp must not be None when initializing snapshot messages.")
+            timestamp = content["timestamp"]
+
+        return super(WazirxOrderBookMessage, cls).__new__(
+            cls, message_type, content, timestamp=timestamp, *args, **kwargs
+        )
+
+    @property
+    def update_id(self) -> int:
+        if self.type in [OrderBookMessageType.DIFF, OrderBookMessageType.SNAPSHOT]:
+            return int(self.timestamp * 1e3)
+        else:
+            return -1
+
+    @property
+    def trade_id(self) -> int:
+        if self.type is OrderBookMessageType.TRADE:
+            return int(self.timestamp * 1e3)
+        return -1
+
+    @property
+    def trading_pair(self) -> str:
+        if "trading_pair" in self.content:
+            return self.content["trading_pair"]
+        elif "instrument_name" in self.content:
+            return self.content["instrument_name"]
+
+    @property
+    def asks(self) -> List[OrderBookRow]:
+        asks = map(self.content["asks"], lambda ask: {"price": ask[0], "amount": ask[1]})
+
+        return [
+            OrderBookRow(float(price), float(amount), self.update_id) for price, amount in asks
+        ]
+
+    @property
+    def bids(self) -> List[OrderBookRow]:
+        bids = map(self.content["bids"], lambda bid: {"price": bid[0], "amount": bid[1]})
+
+        return [
+            OrderBookRow(float(price), float(amount), self.update_id) for price, amount in bids
+        ]
+
+    def __eq__(self, other) -> bool:
+        return self.type == other.type and self.timestamp == other.timestamp
```

Comment:
If `__eq__` message is reimplemented, then `__hash__` has to be reimplemented too, and be consistent with the new definition of `__eq__`

Reply:
Thanks, updated.